dashboard/assets_release.go

```diff
@@ -0,0 +1,10 @@
+//+build release
```

Comment:
I'm not convinced this is the most elegant approach but here is the reasoning behind it.

1. Ideally we don't have to run an optimized UI build every time we make a change to the dashboard. At present the build takes around 45 seconds, which while not bad, will likely increase to 5-10 minutes as the codebase grows.
As such I don't think it is viable to have the generated assets committed to the repo. It would sort of be like packaging sensu for each platform each time we committed.
1. Ideally a user should still be able to compile the backend without having to install nodejs and yarn.
1. Ideally a user should be able to `go get` sensu backend.

Of course these ideals conflict to some degree. Eg. if we go with the good developer experience and don't commit the build artifacts to the repo then the build cannot succeed until `go generate` has run; conversely if we do commit it, we are damning ourselves to frequently re-building and commit the dashboard builds.

To avoid this, the default behavior of this package is to expose an "empty" implementation of `http.FileSystem`. In this case the dashboard will be missing, however, the code will still compile.

When the developer / user is ready to build the dashboard they can simply run:

```
go generate -tags release
go build -tags release
```

Or simply run the build script.